### DISCUSSION TITLE
Vite loading css

### DISCUSSION THREAD

**Discussion by tombohub** [remix-run/remix#8777] [General]
For those who had error: [RollupError]: "default" is not exported by './index.css' imported by "app/root.tsx". 

Don't do 

```javascript
import { LinksFunction } from "@remix-run/node";

import styles from "./index.css";

export const links: LinksFunction = () => [{ rel: "stylesheet", href: styles }];
```

Just do this:

```javascript
import "./index.css";
```

**Comment by akamfoad** [comment]
You need to add `?url` to the end of your CSS import so Vite knows you want `styles` to be the URL of the file, not the CSS content.

Previously Remix handled this out of the box with their compiler, but now they're building on top of Vite so they can focus on Remix features and defer these matters to Vite.

Your code should work correctly like this:

```diff
import { LinksFunction } from "@remix-run/node";

-import styles from "./index.css";
+import styles from "./index.css?url";

export const links: LinksFunction = () => [{ rel: "stylesheet", href: styles }];
```

**Reply by tombohub** [reply]
yes that works too. Is this anything different? The result is the same

**Reply by akamfoad** [reply]
From [Vite docs](https://vitejs.dev/guide/features.html#css)
> Importing .css files will inject its content to the page via a <style> tag with HMR support.

Default export is no longer a URL in [Vite 5](https://vitejs.dev/guide/features.html#disabling-css-injection-into-the-page)
> NOTE
> Default and named imports from CSS files (e.g import style from './foo.css') are removed since Vite 5. Use the ?inline query instead.

The code you provided seems to want the `styles` to be the URL of the `CSS` file, that is exactly what the [`?url` query](https://vitejs.dev/guide/features.html#static-assets) acheives, when provided `styles` is a URL, pointing to the CSS file that'll end up in the dis folder (or build/client in Remix SPA's case)

**Comment by DisruptLive** [comment]
Importing this way exposes the styles defined to the entire application. How would we import if we want to keep a stylesheet scoped or sandboxed to a specific route or component?

**Reply by akamfoad** [reply]
You could import the stylesheet with `?url` postfix which returns a URL, return an object in the `links` function, this way Remix fetches and mounts the styles when the route navigated to, and unmounts the styles when the route is navigated away from.
